### DISCUSSION TITLE
Action Required: Fix Renovate Configuration (JSONata)

### DISCUSSION THREAD

**Discussion by dafyddj** [renovatebot/renovate#31871] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

An "Action Required" issue was opened in my repository on a configuration which worked fine until today.
The only change I'm aware of is the Renovate GitHub app is now running version 38.114.0.

```
There is an error with this repository's Renovate configuration that needs to be fixed. As a precaution, Renovate will stop PRs until it is resolved.

Location: .github/renovate.json5
Error type: The renovate configuration file contains some invalid settings
Message: Invalid JSONata expression for customDatasources: Expected ":", got "}"
```

The relevant configuration seems to be:
```
"firefox": {
        "defaultRegistryUrlTemplate": "https://product-details.mozilla.org/1.0/firefox_versions.json",
        "transformTemplates": [
          "{\"releases\": [{ \"version\": $.{{packageName}} }] }"
        ],
      },
```

The original repository is: https://github.com/saltstack/salt-winrepo-ng

Is there really an error in my configuration?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment]
It seems like a invalid JSONata expression, I checked [here ](https://try.jsonata.org/)and got the same error.

~What was your renovate version before the one you mentioned. Because better logging for JSONata expressions was added in [this ](38.95.4)PR, which wwas released in 38.95.4~

Correction: I confused logging for validation, my bad :)

**Reply by rarkins** [reply]
Validation was added

**Reply by dafyddj** [reply]
Note I am using templating here `{{packageName}}`. The validation seems to be done before the template is replaced.

**Reply by rarkins** [reply]
@secustor so you think we need to revert the validation? Or maybe compile it (which will result in empty bit hopefully valid strings)

**Reply by secustor** [reply, marked as answer]
We should revert, I don't think simply removing templates will work everywhere.

**Reply by rarkins** [reply]
Fix PR: https://github.com/renovatebot/renovate/pull/31878

**Reply by dafyddj** [reply]
Thanks for the quick resolution.

**Reply by rarkins** [reply]
FYI the hosted app isn't updated automatically, so you won't see the issue closed until after that happens plus it runs on your repo once
